.github/workflows/trivy.yaml

```
@@ -0,0 +1,50 @@
+---
+name: trivy
+
+on:
+  workflow_call:
+    inputs:
+      image-ref:
+        required: true
+        type: string
```

Optional, in case you would like to set a default:
```suggestion
    inputs:
      image-ref:
        default: "ghcr.io/${{ github.repository }}:${{ github.sha }}"
        required: false
        type: string
```

---

Ohhh good idea. But, will it be useful for our case ?

---

Depends on how you see it, default values are always good when you can have them.
But it's better to have no default value than a bad one

In this case this would remove the need of having those two lines in the `build` workflow:
```yaml
    with:
       image-ref: 'ghcr.io/${{ github.repository }}:${{ github.sha }}'
```